crates/bevy_render/src/view/window.rs

```diff
@@ -171,7 +171,7 @@ pub fn prepare_windows(
                 render_instance.create_surface(&window.handle.get_handle())
             });
 
-        let swap_chain_descriptor = wgpu::SurfaceConfiguration {
+        let create_swap_chain_descriptor = || wgpu::SurfaceConfiguration {
```

Comment:
```suggestion
        // Creates a closure to avoid calling this logic unnecessarily 
        let create_swap_chain_descriptor = || wgpu::SurfaceConfiguration {
```